diff --git a/build/cjs/logs/console-integration.js b/build/cjs/logs/console-integration.js
index 32383d1cd276d2f15a2406e8096104ac32eb76ce..eecebba0031e7b17a5a019ea97f8c5a5167a5dc3 100644
--- a/build/cjs/logs/console-integration.js
+++ b/build/cjs/logs/console-integration.js
@@ -35,6 +35,10 @@ const _consoleLoggingIntegration = ((options = {}) => {
         const firstArg = args[0];
         const followingArgs = args.slice(1);
 
+        const secondArg = args[1];
+        const timestamp = secondArg && typeof secondArg === 'object' ? secondArg.timestamp : undefined;
+        const timestampInSeconds = timestamp ? timestamp / 1000 : undefined;
+
         if (level === 'assert') {
           if (!firstArg) {
             const assertionMessage =
@@ -53,9 +57,12 @@ const _consoleLoggingIntegration = ((options = {}) => {
         const attributes = {
           ...DEFAULT_ATTRIBUTES,
           ...(shouldGenerateTemplate ? utils.createConsoleTemplateAttributes(firstArg, followingArgs) : {}),
+          patchSource: 'cjs',
+          timestampRecordedByApp: timestampInSeconds || '-',
         };
 
         internal._INTERNAL_captureLog({
+          timestamp: timestampInSeconds,
           level: isLevelLog ? 'info' : level,
           message: utils.formatConsoleArgs(args, normalizeDepth, normalizeMaxBreadth),
           severityNumber: isLevelLog ? 10 : undefined,
diff --git a/build/cjs/logs/internal.js b/build/cjs/logs/internal.js
index c297a85b44e2b85f77c10119fa5ecfa431b2863d..b446baacbf40c6f6472541202da9f1cfb3de565b 100644
--- a/build/cjs/logs/internal.js
+++ b/build/cjs/logs/internal.js
@@ -151,7 +151,7 @@ function _INTERNAL_captureLog(
   const { level, message, attributes: logAttributes = {}, severityNumber } = log;
 
   const serializedLog = {
-    timestamp: time.timestampInSeconds(),
+    timestamp: beforeLog.timestamp ?? time.timestampInSeconds(),
     level,
     body: message,
     trace_id: traceContext?.trace_id,
diff --git a/build/cjs/utils/time.js b/build/cjs/utils/time.js
index d8e7fbd444c2c7a83693a5dfc8fea6f7ea74ff3c..9979cca761fe2a7428cb06aeaf4e9aa815b6cc87 100644
--- a/build/cjs/utils/time.js
+++ b/build/cjs/utils/time.js
@@ -47,21 +47,11 @@ function createUnixTimestampInSecondsFunc() {
   };
 }
 
-let _cachedTimestampInSeconds;
-
 /**
- * Returns a timestamp in seconds since the UNIX epoch using either the Performance or Date APIs, depending on the
- * availability of the Performance API.
- *
- * BUG: Note that because of how browsers implement the Performance API, the clock might stop when the computer is
- * asleep. This creates a skew between `dateTimestampInSeconds` and `timestampInSeconds`. The
- * skew can grow to arbitrary amounts like days, weeks or months.
- * See https://github.com/getsentry/sentry-javascript/issues/2590.
+ * Returns a timestamp in seconds since the UNIX epoch using Date APIs.
  */
 function timestampInSeconds() {
-  // We store this in a closure so that we don't have to create a new function every time this is called.
-  const func = _cachedTimestampInSeconds ?? (_cachedTimestampInSeconds = createUnixTimestampInSecondsFunc());
-  return func();
+  return dateTimestampInSeconds();
 }
 
 /**
diff --git a/build/esm/logs/console-integration.js b/build/esm/logs/console-integration.js
index 52cbeb4d26b2eaa16151b4ec92e067b841165254..d2671fec204c3949b5f228c57e9209a05e100b70 100644
--- a/build/esm/logs/console-integration.js
+++ b/build/esm/logs/console-integration.js
@@ -33,6 +33,10 @@ const _consoleLoggingIntegration = ((options = {}) => {
         const firstArg = args[0];
         const followingArgs = args.slice(1);
 
+        const secondArg = args[1];
+        const timestamp = secondArg && typeof secondArg === 'object' ? secondArg.timestamp : undefined;
+        const timestampInSeconds = timestamp ? (timestamp / 1000) : undefined;
+
         if (level === 'assert') {
           if (!firstArg) {
             const assertionMessage =
@@ -51,9 +55,12 @@ const _consoleLoggingIntegration = ((options = {}) => {
         const attributes = {
           ...DEFAULT_ATTRIBUTES,
           ...(shouldGenerateTemplate ? createConsoleTemplateAttributes(firstArg, followingArgs) : {}),
+          patchSource: 'esm',
+          timestampRecordedByApp: timestampInSeconds || '-',
         };
 
         _INTERNAL_captureLog({
+          timestamp: timestampInSeconds,
           level: isLevelLog ? 'info' : level,
           message: formatConsoleArgs(args, normalizeDepth, normalizeMaxBreadth),
           severityNumber: isLevelLog ? 10 : undefined,
diff --git a/build/esm/logs/internal.js b/build/esm/logs/internal.js
index 76a57455bf0de1b7a12f62e0a8f24b1be81b6281..2da8e60fa3c461f7797fd057ff442f9575dfc749 100644
--- a/build/esm/logs/internal.js
+++ b/build/esm/logs/internal.js
@@ -149,7 +149,7 @@ function _INTERNAL_captureLog(
   const { level, message, attributes: logAttributes = {}, severityNumber } = log;
 
   const serializedLog = {
-    timestamp: timestampInSeconds(),
+    timestamp: beforeLog.timestamp ?? timestampInSeconds(),
     level,
     body: message,
     trace_id: traceContext?.trace_id,
diff --git a/build/esm/utils/time.js b/build/esm/utils/time.js
index 7709a2b5d8e7dea2f2eeb87bd7db389d5ef52179..2e4c47c1d69f3e68809118825f58201759a62528 100644
--- a/build/esm/utils/time.js
+++ b/build/esm/utils/time.js
@@ -45,21 +45,11 @@ function createUnixTimestampInSecondsFunc() {
   };
 }
 
-let _cachedTimestampInSeconds;
-
 /**
- * Returns a timestamp in seconds since the UNIX epoch using either the Performance or Date APIs, depending on the
- * availability of the Performance API.
- *
- * BUG: Note that because of how browsers implement the Performance API, the clock might stop when the computer is
- * asleep. This creates a skew between `dateTimestampInSeconds` and `timestampInSeconds`. The
- * skew can grow to arbitrary amounts like days, weeks or months.
- * See https://github.com/getsentry/sentry-javascript/issues/2590.
+ * Returns a timestamp in seconds since the UNIX epoch using Date APIs.
  */
 function timestampInSeconds() {
-  // We store this in a closure so that we don't have to create a new function every time this is called.
-  const func = _cachedTimestampInSeconds ?? (_cachedTimestampInSeconds = createUnixTimestampInSecondsFunc());
-  return func();
+  return dateTimestampInSeconds();
 }
 
 /**
